equinor/fusion-app-resources
# Aligned Options endpoint for Department with the related endpoints

**INFO**
When a user have DelegatedResposibility, they have "same" acces as if they wherer ReseourceOwner.
Having  DelegatedResposibility means that the user have the AccessRoles.ResourceOwner for a speicified deparment. ( same as ResoruceOwner) 



- [ ] High impact

**Description of work:**
Added 
The other endpoint    [HttpOptions("/projects/{projectIdentifier}/requests/{requestId}")]  already works as expected to be and   returns PATCH,DELETE,GET

So the work done have been to apply the similare Rules [HttpOptions("/departments/{departmentPath}/resources/requests/{requestId}")]  based on testing it should reunt Get and PATCH, DELETE  are not needed as it is not Authorized by the delete endpoint 

AB33477

**Testing:**
- [X] Can be tested
- [ ] ~~Automatic tests created / updated~~
- [X] Local tests are passing

Have been tested by Tony hagen. he thinks it appeasr to be workingas expected. 

**How to test**
When accesing a deparment where user have delegatedresposibility, user should be able to see the Assig/Unnasign and Handle Resuest Buttons, if request are Created and Propsoed ( in pending state) 

**adding delegated acces can be done using this endpoint:** POST
https://resources-api.ci.fusion-dev.net/departments/:fulldepartment/delegated-resource-owner/
BODY
{
  "responsibleAzureUniqueId": "USERGUID",
  "dateTo": "2023-12-31T23:15:04.525Z",
  "dateFrom": "2022-09-30T11:15:04.525Z"
}
**removing delegated acces :** DELETE
https://resources-api.ci.fusion-dev.net/departments/:fulldepartment/delegated-resource-owner/_USERGUID

**My unit's Reqeusts** ( this is where the probem was) Deparment endpoint
![image](https://user-images.githubusercontent.com/3900367/203299004-50dacbd1-c3ff-4f10-98c3-7ce84a8099ea.png)

**Unassigned** Request  ( already returned the neccesary Options ) Projects Endpoint
![image](https://user-images.githubusercontent.com/3900367/203299318-65c333f8-4135-47d6-8fbb-3549f61548b8.png)

So both these OPTIONS endpoints should return same Options. PATCH,DELETE,GET

**Checklist:**
- [ ] ~~Considered automated tests~~
- [ ] ~~Considered updating specification / documentation~~
- [X] Considered work items 
- [ ] ~~Considered security~~
- [X] Performed developer testing
- [X] Checklist finalized / ready for review
